src/app/routes/accountRecoveryRoute.js

```diff
@@ -6,76 +6,110 @@ const uuidv1 = require('uuid/v1')
+      try {
+        var reqUserId = _.get(req.body, 'request.userId')
+        var reqValidator = _.get(req, 'body.request.validator')
+        var decodedValidator = isValidRequest(reqValidator);
+        // checking only for the userID from request and from the decoded object.
+        if ((decodedValidator['userId']) && (reqUserId === decodedValidator['userId'])) {
+          next();
+        } else {
+          logger.error({
+            msg: 'unauthorized'
+          });
+          res.status(401).send({ "id": "api.reset.password", "ver": "v1", "ts": dateFormat(new Date(), 'yyyy-mm-dd HH:MM:ss:lo'), "params": { "resmsgid": null, "msgid": uuidv1(), "err": null, "status": "unauthorized", "errmsg": null }, "responseCode": "UNAUTHORIZED", "result": { "response": "unauthorized" } })
+        }
+      } catch (err) {
         logger.error({
-          msg: 'unauthorized'
+          URL: req.url,
+          body: JSON.stringify(req.body),
+          msg: 'portal - otp verification failed',
+          error: JSON.stringify(err)
         });
-        res.status(401).send({"id":"api.reset.password","ver":"v1" ,"ts":dateFormat(new Date(), 'yyyy-mm-dd HH:MM:ss:lo'),"params":{"resmsgid":null,"msgid": uuidv1(),"err":null,"status":"unauthorized","errmsg":null},"responseCode":"UNAUTHORIZED","result":{"response":"unauthorized"}})
-      } else {
-        next()
       }
     },
     proxy(envHelper.learner_Service_Local_BaseUrl, {
       proxyReqOptDecorator: proxyUtils.decorateRequestHeaders(),
       proxyReqPathResolver: (req) => {
         return '/private/user/v1/password/reset'; // /private/user/v1/reset/password
       }
-  }))
+    }))
 
   app.all('/learner/otp/v1/verify',
-    bodyParser.urlencoded({ extended: false }), bodyParser.json({ limit: '10mb' }), 
+    bodyParser.urlencoded({ extended: false }), bodyParser.json({ limit: '10mb' }),
     proxy(envHelper.LEARNER_URL, {
       proxyReqOptDecorator: proxyUtils.decorateRequestHeaders(),
       proxyReqPathResolver: (req) => {
         return require('url').parse(envHelper.LEARNER_URL + req.originalUrl.replace('/learner/', '')).path
       },
       userResDecorator: (proxyRes, proxyResData, req, res) => {
         try {
-            proxyUtils.addReqLog(req);
-            const data = JSON.parse(proxyResData.toString('utf8'));
-            if (data.responseCode === 'OK') {
-              req.session.otpVerifiedFor = req.body;
-              var validator = getEncyptedQueryParams({'key':req.body.request.key});
-              data['validator'] = validator;
+          proxyUtils.addReqLog(req);
+          const data = JSON.parse(proxyResData.toString('utf8'));
+          if (data.responseCode === 'OK') {
+            req.session.otpVerifiedFor = req.body;
+            const encrypt = {
+              key: req.body.request.key
+            }
+            if (req.body.request.userId) {
+              encrypt['userId'] = req.body.request.userId
```

Comment:
put the generic name as Id:.
So others can't understand what is this id(in the encrypted data);